File: nodejs/v1_soap_client.js
const http = require('http');
const https = require('https');


const server = http.createServer((req, res) => {
    // Parsear la URL de forma nativa para obtener el parámetro 'n'
    const url = new URL(req.url, `http://${req.headers.host}`);
    const n = url.searchParams.get('n');

    if (!n) {
        res.writeHead(400, { 'Content-Type': 'text/plain; charset=utf-8' });
        return res.end('Por favor, proporcione el parámetro n en la URL.');
    }


    const xmlPayload = `<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <NumberToWords xmlns="http://www.dataaccess.com/webservicesserver/">
      <ubiNum>${n}</ubiNum>
    </NumberToWords>
  </soap:Body>
</soap:Envelope>`;

    // Opciones para la petición HTTPS nativa al WSDL
    const options = {
        hostname: 'www.dataaccess.com',
        port: 443,
        path: '/webservicesserver/NumberConversion.wso',
        method: 'POST',
        headers: {
            'Content-Type': 'text/xml; charset=utf-8',
            'Content-Length': Buffer.byteLength(xmlPayload)
        }
    };

    // Consumo SOAP
    const soapReq = https.request(options, (soapRes) => {
        let data = '';
        
        // Recibir los fragmentos de datos
        soapRes.on('data', (chunk) => {
            data += chunk;
        });

        // Al terminar de recibir la respuesta
        soapRes.on('end', () => {
            const match = data.match(/<m:NumberToWordsResult>(.*?)<\/m:NumberToWordsResult>/);
            const result = match ? match[1].trim() : "Error en la respuesta SOAP";
            
            // Responder al navegador web
            res.writeHead(200, { 'Content-Type': 'text/plain; charset=utf-8' });
            res.end(result);
        });
    });

    soapReq.on('error', (e) => {
        res.writeHead(500, { 'Content-Type': 'text/plain; charset=utf-8' });
        res.end('Error interno del servidor al conectar con el servicio SOAP.');
    });

    soapReq.write(xmlPayload);
    soapReq.end();
});

server.listen(8000, () => {
    console.log('Servidor Node.js V1 (SOAP) activo en http://localhost:8000/v1_soap_client.js');
});

File: golang/v1_soap_client.go
package main

import (
	"bytes"
	"fmt"
	"io"
	"net/http"
	"regexp"
)

func main() {
	http.HandleFunc("/v1_soap_client", func(w http.ResponseWriter, r *http.Request) {
		n := r.URL.Query().Get("n")
		if n == "" {
			w.WriteHeader(http.StatusBadRequest)
			w.Header().Set("Content-Type", "text/plain; charset=utf-8")
			fmt.Fprint(w, "Por favor, proporcione el parametro n en la URL.")
			return
		}

		xmlPayload := fmt.Sprintf(`<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <NumberToWords xmlns="http://www.dataaccess.com/webservicesserver/">
      <ubiNum>%s</ubiNum>
    </NumberToWords>
  </soap:Body>
</soap:Envelope>`, n)

		req, err := http.NewRequest("POST", "https://www.dataaccess.com/webservicesserver/NumberConversion.wso", bytes.NewBufferString(xmlPayload))
		if err != nil {
			w.WriteHeader(http.StatusInternalServerError)
			fmt.Fprint(w, "Error interno al crear la peticion")
			return
		}

		req.Header.Set("Content-Type", "text/xml; charset=utf-8")

		client := &http.Client{}
		resp, err := client.Do(req)
		if err != nil {
			w.WriteHeader(http.StatusInternalServerError)
			fmt.Fprint(w, "Error interno al conectar con SOAP")
			return
		}
		defer resp.Body.Close()

		body, err := io.ReadAll(resp.Body)
		if err != nil {
			w.WriteHeader(http.StatusInternalServerError)
			fmt.Fprint(w, "Error al leer la respuesta")
			return
		}

		re := regexp.MustCompile(`<m:NumberToWordsResult>(.*?)</m:NumberToWordsResult>`)
		match := re.FindStringSubmatch(string(body))

		if len(match) > 1 {
			w.Header().Set("Content-Type", "text/plain; charset=utf-8")
			fmt.Fprint(w, match[1])
		} else {
			w.WriteHeader(http.StatusInternalServerError)
			fmt.Fprint(w, "Error en la respuesta SOAP")
		}
	})

	http.ListenAndServe(":8000", nil)
}